### ISSUE TITLE
Yakyak not reporting current user as available on other clients

### ISSUE THREAD

**Issue by randymorris** [yakyak/yakyak#167]
Using Yakyak Version 0.37.4 (1.2.0) I am getting complaints from coworkers that I am not showing up as available on official Google clients even though I am logged in and chatting with other people.

I don't believe this is specific to 1.2.0 as this has been mentioned once or twice in the past.  I had ignored it then thinking it was just an issue with their client lagging in presence updates, but the last few times have been after I had been signed in via Yakyak for hours.

Can anyone else reproduce?

**Comment by vgavrila** [comment]
I can recreate. But is this a bug or an enhancement?

**Comment by randymorris** [comment]
IMHO it's a bug.  Until there is a way to set your status I think you should be set to available.  As it stands, I have to run two different clients just to let people know I'm online.

**Comment by davibe** [comment]
It's a bug on user's perspective. But since the code never supported this it's an enhancement for the project.

**Comment by anevilsithlord** [comment]
Are there plans to include User Status in a future release?

**Comment by arnriu** [comment]
@anevilsithlord yes definitely. It's on my list unless someone does it before ;)

**Comment by Stanko** [comment]
I can recreate as well.
This would be really nice to get into the next release.

**Comment by davibe** [comment]
so, do it and send us a pull request!

**Comment by Stanko** [comment]
@davibe I actually already took a look, but I'm not sure where is the problem. I would appreciate advice where to start.

**Comment by etrulls** [comment]
Is this still the expected behaviour? I keep getting messages from coworkers that I'm showing as offline.

**Comment by NvrBst** [comment]
Also seeing this. The YakYak client is setting myself as "Invisible", so no one can see I am online. I'm using the latest YakYak Client (1.4.1). It seems to me that the client should show as "Available" or "Online" by default...

Everyone using YakYak is fine with never showing as online to anyone? This seems like a pretty big bug to me... I'm not too sure about anyone else, but assuming people are using only YakYak, not having anyone able to see if you're online ever (and thus probably not sending you messages)... then what's the point in having the messaging client up then in the first place?

If you want "Invisible" that's fine... but it should definitely not be the default behavior...

**Comment by davibe** [comment]
I see myself as logged in (green) when using desktop hangout website from another user.

<img width="192" alt="screenshot 2017-01-12 18 15 06" src="https://cloud.githubusercontent.com/assets/123929/21908292/6de5687e-d913-11e6-98b3-1b81d8066a90.png">

**Comment by NvrBst** [comment]
You may be logged into another Hangouts client (like on your phone or opened a gmail window).

Or you recently used some other client within the last 10 mins'ish, closed it, then the "Online Status" remains for a while.

Else if you start talking with someone on yakyak, it will change your status to online (i believe). But when you stop talking with someone, or keep yakyak minimised for a while (and you do not have another client open, like your phone keeping your online status updated), it'll just show you as Offline in a bit, even though YakYak is open on the desktop.

**Comment by averissimo** [comment]
Yeah, that's because of "presence" timeout of 20 seconds I think everytime a message is sent.

I can confirm the issue in my setup.

Screenshot from gmail's web interface
![image](https://cloud.githubusercontent.com/assets/211358/21954358/3fa54db0-da46-11e6-80f5-cb50542f5e5f.png)

**Comment by davibe** [comment]
Actually I logged off all clients, waited a few hours, and then tried. I will try again

**Comment by NvrBst** [comment]
Yea, basically you didn't wait long enough (or you have mouse activity). So for me, if you make sure you are logged off all other clients. Only have yakyak logged in. And then keep it minimised or (minimised to tray - either or) for a while. Then open an "incognito window", and go to hangouts web on your other account, you will see your main account is "Offline".

Now you can open the YakYak window, and you will still be "Offline". As soon as you move your mouse anywhere over the YakYak Client, then it'll show as "Online" on your other account. So you don't have to message someone, you just have to have mouse activity within the YakYak Client.

You probably opened the YakYak client and had mouse activity in it sometime. That said, most of the time YakYak is minimised, and thus most of the time you'll be appearing as "Offline" to your contacts.


Also "Toggle Minimized to Tray" seems to not do what I assume. I assume it's a Toggle, so when turned on, if you minimise, it'll minimise to tray. Instead if you click it, it'll minimise to tray instantly, no checkmark or anything will ever appear beside it. Minimize always will minimise to taskbar no matter what you do with that option.
Not too sure it that's intended behaviour or not. If so, it should be renamed to not have Toggle, which implies it'll affect the "minimise function", which currently it doesn't. Maybe "Send to Tray".

The "Close to Tray" works as expect, where if turned on, the "Close" function sends it to tray. The Minimize one doesn't.

**Comment by averissimo** [comment]
I would like to correct my previous comment, as I didn't have the option below set. If I tick that option true, then my user appears as online in YakYak immediately

![image](https://cloud.githubusercontent.com/assets/211358/22376747/26f17d0a-e4a7-11e6-9c8d-e8568d9387af.png)

I am logging out of a test account right now (gmail + yakyak) and will re-login via yakyak as soon as the 'green dot' disappears.

**Comment by averissimo** [comment]
I did a test and reproduced what you said using Luís Castro account, description below.

@NvrBst I guess this is not a bug, but it was a design option.. Idle users show as not available even though YakYak is open.

Should presence be set every X minutes every time the app is open (or attached to some connection monitoring event)

Or should this be left alone and the available only set once the user is active?

------

Test:

16h20: Before opening YakYak

![image](https://cloud.githubusercontent.com/assets/211358/22378161/8db3806a-e4ac-11e6-98af-f59a1c9f3752.png)

After opening YakYak (small 2-3 seconds delay)

![image](https://cloud.githubusercontent.com/assets/211358/22378187/a22c4e14-e4ac-11e6-8094-02ad35a8f833.png)

30minutes later user appeared as offline

![image](https://cloud.githubusercontent.com/assets/211358/22378967/4a1f0b0a-e4af-11e6-8b0b-ccd69193e939.png)

**Comment by averissimo** [comment]
Maybe the phrasing in "Toggle minimise to tray" is not the best, the behaviour is as you described, immediatly minimizing to tray, or showing if it is minimized.

Can you propose something that better describes this behaviour? :)

**Comment by NvrBst** [comment]
I would take the word minimise out, as it describes actions involving the "minimise button", I would do something more like "**Show/Hide to tray**".
Then the "Close to tray" could be renamed to "**Toggle 'Close to tray'**" which describes more what that option is doing.

>  it was a design option.. Idle users show as not available even though YakYak is open.

As for the idle users showing as offline. I would say it would be best to mimic what other official Hangout Clients do, (Gmail Web / Hangouts Web / Hangouts Desktop Chrome App). They all keep the user as "Online" when the client is open. For the Hangouts Desktop Chrome App, I would just keep the App minimised in the taskbar, and never open it. It would keep me as "Online" all day long.
Perhaps YakYak should just set the presence every 5 or 10 mins, so the availability is always "Online". Confirming, when you move your mouse over the client it doesn't update the "Active 2h Ago" part (that is set for the last txt you sent? it only updates the "Online Status"). If so, yea YakYak should always be setting the presence while the client is open.

**Comment by averissimo** [comment]
#571 solves the presence issue

I don't like 'Show/Hide to tray', 'Show/Hide Window', should be better

But I'm not the best person to comment on this, I have a custom version that always brings window to focus

(the choice must be a good a definite one, as this will require a change in translations)

can you open a new issue if you feel strongly about this? (as this is somewhat offtopic for this one)

**Comment by NvrBst** [comment]
Awesome! Thanks for fixing that. I see you just updated with a pre-release of 1.4.2 but this commit just missed getting in that pre-release! hehe. I'll have to wait but will test ASAP!. Thanks again.

Yea updating the translation may not be worth it. A lot of work. And once people figure out what the option does, it doesn't matter anymore. It's not a bit deal for now. If you do change it, I would mimic what Mozilla software does (Can take their translations too for the option). And they are usually named:
"Hide to tray".
"Hide to tray on start"
"Hide to tray on close"
"Hide to tray on minimize"

"Hide/Show Window" works too. doesn't matter. But yea not important right now I would say. Thanks again for fixing the setprescence!!

**Comment by averissimo** [comment]
@NvrBst you're very welcome, waiting on another developer review to merge and update the pre-release, but as we are only 2 _(at the moment)_ it might take a few days :)

Just realised I should have checked those generic strings before asking for translations lol

I guess if anyone feels strong enough and creates a pull request they will change and eventually trickle down to all translations

And by anyone it can be you :)

I'm closing this issue, any discussion directly related to this issue can be done at the pull request #571

**Comment by NvrBst** [comment]
Heya!

I'm still seeing this issue. I have upgraded to the latest pre-release (v1.4.2 - 9ade528) released 1 day ago.  I confirmed the change you made to set the presence on every _"update:connection"_ is in the code on my system:
```
ui/source/controller.js
      if (connection.state === connection.CONNECTED) {
        later(function() {
          return action('lastActivity');
        });
```
I deleted all the other YakYak's on my system, and ran this one. However, when I leave YakYak in the "Closed to Tray" state, my online status is set to Offline after a while. I have timed how long this takes, by moving my mouse around the YakYak client, then closing it to tray and seeing when it sets me to offline. It does it in exactly 15 minutes and 10 seconds. You have to log in and out of hangouts web interface to have the client refresh the online status of the list quicker. Else the web client doesn't update the people's status as often unless you relog in.

How often does this _"update:connection"_ happen in a _"connection.CONNECTED"_ state? Can the connection state be changed from "CONNECTED" to something other after a while? i.e. "IDLE" or "CLOSED" if no activity has been detected for awhile? Then it stops setting the presence? Maybe it doesn't go into this state every 15 mins?

Maybe setting _lastActivity_ periodically may be better? Every 10-minutes to stay within the 15-minute window?
